src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessor.cs

```diff
@@ -111,5 +113,18 @@ async Task ProcessInitializingHandler(PartitionInitializingEventArgs initArgs)
                 processor.PartitionInitializingAsync -= ProcessInitializingHandler;
             }
         }
+
+        private (Exception exception, string reason) ProcessErrorEventException(Exception exception)
+        {
+            switch (exception)
+            {
+                case EventHubsException eventHubsException:
+                    return (eventHubsException, $"{_eventHubErrorPrefix}{eventHubsException.Reason}");
```

Comment:
@wi-y Because we are logging this as a metric above will these by exposed to PaaS users as is or is there additional wiring up needed to facilitate that?

Reply:
Yeah it will be exposed as an error metric to PaaS users without additional wiring

Reply:
Thanks for confirming Will, that is what I thought.  @pallar-ms before we commit I think we need to make sure these are the values we customers & users to see in the product.  Do you have some examples that would be generated?

Reply:
Couple examples - 
* EventHubErrorResourceNotFound - frrom the EventHubsException when the event hub provided is not found.
* EventHubErrorHostNotFound - from the SocketException, when the event hub fqdn is invalid.
* EventHubErrorGeneralError - catch all for other exceptions. 
Unfortunately from the documentation, I don't see all the expected exception types that we could specifically handle.

Reply:
Okay thanks for the examples Pallavi.  My one concern here is we are relying on the Reason imbedded in the Event Hub exception to form the name we are emitting.  If there is change in the Event Hub library if we could generate different telemetry for the same issue.  I would like to instead have constants we control for these categories for consistency such scenario.  Does that make sense?  With that change other suggestions like potentially refactoring into a dedicated exception type can be deferred.

Reply:
Yeah I sort of agree, I had the same concern at first of using Reason in case things changed, however from debugging perspective it also made sense to have it reflect the event hub codes as is.
* If the suggestion was to have a mapping b/w the event hub "reasons" and our custom "reasons", we'd miss out on updates that way when new reason codes are created.
* Else we can have a generic reason string but also capture the event hub reason some where although I don't think we have a spare field for that.

Reply:
We can use the ILogger and log the original internally for us.  The main concern is these names will be exposed to customers that are using Azure Monitor integration.  So if we release this and a customer builds alerts off this information but then later there is a change in the supporting library that could change the name we will break their alerts.

Reply:
Got it, makes sense!